File: sdks/ios/Sources/XMTPiOS/SignedData.swift
import Foundation

public struct SignedData {
	/// The raw signature bytes.
	public let rawData: Data

	/// The public key used for signing (required for Passkeys).
	public let publicKey: Data?

	/// The WebAuthn authenticator data (only for Passkeys).
	public let authenticatorData: Data?

	/// The WebAuthn client data JSON (only for Passkeys).
	public let clientDataJson: Data?

	public init(rawData: Data, publicKey: Data? = nil, authenticatorData: Data? = nil, clientDataJson: Data? = nil) {
		self.rawData = rawData
		self.publicKey = publicKey
		self.authenticatorData = authenticatorData
		self.clientDataJson = clientDataJson
	}
}

File: sdks/android/jvm/src/main/java/org/xmtp/kotlin/SignedData.kt
package org.xmtp.kotlin

data class SignedData(
    val rawData: ByteArray,
    val publicKey: ByteArray? = null, // Used for Passkeys
    val authenticatorData: ByteArray? = null, // WebAuthn metadata
    val clientDataJson: ByteArray? = null, // WebAuthn metadata
)
